### PR TITLE
config: set default region for EC2 SD

### PR DESCRIPTION
**Pull request by mtanda**
If Prometheus is running on EC2, most of the case `region` might be same region where Prometheus is running.
I want to set default region for the case.

### CODE REVIEW

**Comment by mtanda** [comment]
I forgot to mention maintainers.

@brian-brazil Would you check this PR?

File: config/config.go [hunk 1]
```diff
@@ -1126,7 +1128,13 @@ func (c *EC2SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		return err
 	}
 	if c.Region == "" {
-		return fmt.Errorf("EC2 SD configuration requires a region")
+		sess := session.Must(session.NewSession())
```
Comment (fabxc) [inline config/config.go]: Sounds like this could panic if something goes wrong. Better to check potential errors?

**Comment by mtanda** [comment]
> Change seems fine to me in general if that's how AWS clients work with regions in general.

I think..., the `aws-sdk-go` should assure that.

File: config/config.go (hunk 1 above)
Comment (mtanda) [inline config/config.go]: The errors seems to be caused by wrong configuration.
https://github.com/aws/aws-sdk-go/blob/v1.8.35/aws/session/session.go#L100
https://github.com/aws/aws-sdk-go/blob/v1.8.35/aws/session/session.go#L257
https://github.com/aws/aws-sdk-go/blob/v1.8.35/aws/session/session.go#L335
https://github.com/aws/aws-sdk-go/blob/v1.8.35/aws/session/session.go#L339
https://github.com/aws/aws-sdk-go/blob/v1.8.35/aws/session/session.go#L352

But it is better to add error handling, so I add it :-)

**Comment by mtanda** [comment]
@fabxc @brian-brazil How I can proceed this PR?
I should fix something?

**Comment by fabxc** [comment]
My bad, looks all good. Thanks!

**Comment by mtanda** [comment]
Thanks!!

**Pull request merged**